src/hardwareinfo.rs

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::collections::HashMap;
+
+use astarte_sdk::types::AstarteType;
+use serde::Serialize;
+
+use crate::error::DeviceManagerError;
+
+/// get structured data for `io.edgehog.devicemanager.HardwareInfo` interface
+pub fn get_hardware_info() -> Result<HashMap<String, AstarteType>, DeviceManagerError> {
+    let uname = nix::sys::utsname::uname();
+
+    let cpuinfo = procfs::CpuInfo::new()?;
```

Comment:
I think we should move this closer to where we use it. We might move this line to line 42, just before of doing:
`if let Some(f) = cpuinfo.fields.get("model") {`.

Reply:
The same approach should be applied everywhere, unless we have a reason for doing it in a different way.